src/helpers/recompute_zones.ts

```diff
@@ -0,0 +1,386 @@
+    // --> it's a new profile
+    // --> we need to add it
+    profilesStartingPosition.splice(predecessorIndex + 1, 0, value);
+    profiles.set(value, [...profiles.get(profilesStartingPosition[predecessorIndex])!]); // we use the profile of the predecessor to modify it in modifyProfile
+    return searchLeft ? predecessorIndex + 1 : predecessorIndex;
+  }
+  return searchLeft ? predecessorIndex : predecessorIndex - 1;
+}
+
+/**
+ *  Suppose the following        Suppose we want to add          We want to have the
+ *  profile:                     the following zone:             following profile:
+ *
+ *       A B C D E F                  A B C D E F                     A B C D E F
+ *     1    '___'                   1    '   '                      1    '___'
+ *     2    |___|                   2    '___'                      2    |   |
+ *     3    '   '                   3    |   |                      3    |   |
+ *     4    '___'          -->      4    |   |            -->       4    |   |
+ *     6    |   |                   6    |___|                      6    |   |
+ *     7    |___|                   7                               7    |___|
+ *     8                            8                               8
+ *
+ *  the profile for 'C'          the top zone correspond        Here [2, 3, 5, 8] with [3, 7]
+ *  corresponds to:              to 3 and the bottom zone       would be merged into [2, 8]
+ *   ____  ____                  correspond to 6
+ *  [2, 3, 5, 8]                 would be the profile:          The difficulty of modify profile
+ *                                ____                          is to know what must be deleted
+ *  Note that the 'filled        [3, 7]                         and what must be added to the
+ *  zone' are always between                                    existing profile.
+ *  an even index and its
+ *  next index
+ *
+ */
+
+function modifyProfile(
+  profile: number[],
+  zone: UnboundedZone,
+  toRemove: boolean = false
+): undefined {
+  const topValue = zone.top;
+  const bottomValue = zone.bottom === undefined ? undefined : zone.bottom + 1;
+  const newPoints: number[] = [];
+  // Case we want to add a zone to the profile:
+  // - If the top predecessor index is even, it means that the top of the zone is already positioned on a filled zone
+  // so we don't need to add it to the profile. we can keep in reference the index of the predecessor.
+  // - If it is odd, it means that the top of the zone must be the beginning of a filled zone.
+  // so we can keep the index of the top position
+  // Case we want to remove a zone from the profile: it's the opposite of the previous case
+  const topPredIndex = binaryPredecessorSearch(profile, topValue, 0, false);
+  if ((topPredIndex % 2 !== 0 && !toRemove) || (topPredIndex % 2 === 0 && toRemove)) {
+    newPoints.push(topValue);
+  }
+
+  if (bottomValue === undefined) {
+    profile.splice(topPredIndex + 1);
+    profile.push(...newPoints);
```

Comment:
Suppose the following profile on B:C [2,3,5,8] and we want to add the zone B3:C,
the profile on B:C should become [2]

Reply:
I will add tests for that

Reply:
Note that will not impact the final result, but it will impact the intermediate profile. So, for performance reason, it is important to keep these lines.